### PR TITLE
Update template engine

### PR DESCRIPTION
**Pull request by mlorbetske**
Add the build logic to pick up the timestamped store

Template engine changes:
- Fix an issue where C++ literals were having symbols interpolated
- Fix an issue with spaces in parameters to templates causing generation failure with a message about aliases
- Update assembly version numbers to match the CI build

@mikeharder @Eilon @livarcocc @seancpeters

### CODE REVIEW

**Comment by mlorbetske** [comment]
@dotnet-bot test Ubuntu16.10 x64 Debug Build please
@dotnet-bot test RHEL7.2 x64 Release Build please

**Comment by mlorbetske** [comment]
@dotnet-bot test CentOS7.1 x64 Debug Build please
@dotnet-bot test Linux x64 Release Build please
@dotnet-bot test OSX10.12 x64 Release Build please
@dotnet-bot test OpenSUSE42.1 x64 Debug Build please
@dotnet-bot test Ubuntu x64 Release Build please
@dotnet-bot test Ubuntu16.04 x64 Debug Build please

**Comment by livarcocc** [comment]
@MattGertz this is just dependencies flowing into the CLI.

File: src/dotnet/commands/dotnet-new/NewCommandShim.cs [hunk 1]
```diff
@@ -42,24 +43,11 @@ public static int Run(string[] args)
 
         private static ITemplateEngineHost CreateHost()
         {
-            var builtIns = new Dictionary<Guid, Func<Type>>
+            var builtIns = new AssemblyComponentCatalog(new[]
             {
-                { new Guid("0C434DF7-E2CB-4DEE-B216-D7C58C8EB4B3"), () => typeof(RunnableProjectGenerator) },
```
Comment (livarcocc) [inline src/dotnet/commands/dotnet-new/NewCommandShim.cs]: I am so happy right now.

Comment (mlorbetske) [inline src/dotnet/commands/dotnet-new/NewCommandShim.cs]: i aim to please :)

**Pull request merged**